packages/wdio-reporter/src/stats/test.ts

```diff
@@ -87,10 +94,58 @@ export default class TestStats extends RunnableStats {
     fail(errors?: Error[]) {
         this.complete()
         this.state = 'failed'
-        this.errors = errors
 
-        if (errors && errors.length) {
-            this.error = errors[0]
+        const formattedErrors = errors?.map((err: Error) => (
+            (
+                err instanceof AssertionError ||
+                (
+                    typeof (err as AssertionError).expected !== 'undefined' &&
+                    typeof (err as AssertionError).actual !== 'undefined'
+                )
+            )
+                ? this._stringifyDiffObjs(err as AssertionError)
+                : err
+        ))
```

Comment:
This bit of code has some complexity to it. I think it could stand to have a single line comment explaining what it does. Something like, "Checks all errors to check if they're a type of 'AssertionError', and formats it if so. Otherwise, just leaves error as is"